site/app/templates/calendar/Calendar.twig

```diff
@@ -0,0 +1,70 @@
+{% import _self as self %}
+
+<script>
+    let gradeables_by_date = {{ gradeables_by_date|json_encode|raw }};
+    let curr_year = {{ curr_year }};
+    let curr_month = {{ curr_month }};
+    let curr_day = {{ curr_day }};
+
+    window.onload = function () {
+        loadCalendar({{ view_month }}, {{ view_year }});
+    }
+</script>
+
+{# Begin Calendar #}
+<div class="content">
+    <h1>Calendar</h1>
+    <div class="cal-content">
+        <div class="cal-toolbar">
+            {# Replace by js function call #}
+        </div>
+
+        <div class="calendar" style='margin-bottom: 20px' id="full_calendar">
+        </div>
+    </div>
+
+    {# List items in table #}
+    <div class="lst-content">
+        {% if show_only_calendar == 0 %}
+            {% if gradeables_by_section|length > 0 %}
+                <div id="gradeables">
+                {% for section_num, info in gradeables_by_section %}
+                    <div style='margin-bottom: 20px'>
+                        <div id="{{ info.section_id }}" class="course-section-heading">
+                            <div class="course-section-title">
+                                {{ info.title }}
+                                {% if info.subtitle != "" %}
+                                    &nbsp;&nbsp;<em>{{ info.subtitle }}</em>
+                                {% endif %}
+                            </div>
+                        </div>
+
+                        <table class='table table-striped table-bordered persist-area' style='text-align: left'>
+                            <thead>
+                                <tr>
+                                    <th>Course</th>
+                                    <th>Gradeable</th>
+                                    <th>Submission Closes</th>
+                                </tr>
+                            </thead>
+                            <tbody>
+                            {% for id, gradeable in info.gradeables %}
+                                <tr>
+                                    <td>{{ gradeable.course }}</td>
+                                    <td><a href="{{ gradeable.url }}">{{ gradeable.title }}</a></td>
+                                    <td>{{ (gradeable.submission != '') ? gradeable.submission|date('m/d/Y @ h:i A T') : '' }}</td>
```

Comment:
Thanks for this suggestions. Surely a pre-configured date format is important and I didn't know there was a config. I'd like to ask is there a way also to enable this filter in Javascript files？

Reply:
hmm I don't think there's a good way, the PHP DateTime reference has a ton of formatting options and JS doesn't really have an equivalent function. If possible you can construct the date and pass it from the server and embed it to JS. But other than that I think the only way is to construct the date in the expected format.

And above I just chose `getFormat('gradeable') ` as an example. You can add your own 'calendar' format in
`site/app/models/DateTimeFormat`. Getting the same format from this model everywhere in the calendar helps us in case we decide to change the formatting in the future, we only have to change 1 place.